adapters/connectad/connectad.go

```diff
@@ -0,0 +1,209 @@
+package connectad
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type ConnectAdAdapter struct {
+	endpoint string
+}
+
+type connectadImpExt struct {
+	ConnectAd openrtb_ext.ExtImpConnectAd `json:"connectad"`
+}
+
+func NewConnectAdBidder(endpoint string) *ConnectAdAdapter {
+	return &ConnectAdAdapter{
+		endpoint: endpoint,
+	}
+}
+
+func (a *ConnectAdAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	var errs []error
+
+	if errs := preprocess(request); len(errs) > 0 {
+		return nil, append(errs, &errortypes.BadInput{
+			Message: fmt.Sprintf("Error in preprocess of Imp, err: %s", errs),
+		})
+	}
+
+	data, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Error in packaging request to JSON"),
+		}}
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	if request.Device != nil {
+		addHeaderIfNonEmpty(headers, "User-Agent", request.Device.UA)
+		addHeaderIfNonEmpty(headers, "Accept-Language", request.Device.Language)
+		if request.Device.IP != "" {
+			addHeaderIfNonEmpty(headers, "X-Forwarded-For", request.Device.IP)
+		} else if request.Device.IPv6 != "" {
+			addHeaderIfNonEmpty(headers, "X-Forwarded-For", request.Device.IPv6)
+		}
+		if request.Device.DNT != nil {
+			addHeaderIfNonEmpty(headers, "DNT", strconv.Itoa(int(*request.Device.DNT)))
+		} else {
+			addHeaderIfNonEmpty(headers, "DNT", "0")
+		}
+	}
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    data,
+		Headers: headers,
+	}}, errs
+}
+
+func (a *ConnectAdAdapter) MakeBids(bidReq *openrtb.BidRequest, unused *adapters.RequestData, httpRes *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if httpRes.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if httpRes.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Invalid Status Returned: %d. Run with request.debug = 1 for more info", httpRes.StatusCode),
+		}}
+	}
+
+	var bidResp openrtb.BidResponse
+
+	if err := json.Unmarshal(httpRes.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unable to unpackage bid response. Error: %s", err.Error()),
+		}}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
```

Comment:
Please consider setting capacity to `len(bidResp.SeatBid)` to slightly improve memory allocation.